### ISSUE TITLE
Export Class Support

### ISSUE THREAD

**Issue by evanleck** [romainl/ctags-patterns-for-javascript#8]
I'm pretty unfamiliar with the syntax for ctags regex and was wondering if there's a reason that the following wouldn't work?

```js
export class Foo {
  // ...
}
```

I'd expect to be able to reach `Foo` as a class if that makes sense.

**Comment by romainl** [comment]
The pattern for the `Class` kind assumes this structure:

    [BOL][0+ whitespace][class][1+ whitespace][name]

You would need to add another pattern for that one:

    [BOL][0+ whitespace][export][1+ whitespace][class][1+ whitespace][name]

But I'm conflicted about this request as `Foo` would be both a named export *and* a class.

**Comment by evanleck** [comment]
Hm, yeah I see get the conflict. Is that just a semantic difference in ctags or is it more meaningful?

**Comment by evanleck** [comment]
For the pattern, would this make sense?

```
--regex-javascript=/^[ \t]export[ \t]*class[ \t]\{1,\}\([A-Za-z0-9_$]\{1,\}\)/\1/C,Class,Classes/b
```

I just added `export[ \t]` in front of `*class`...

**Comment by romainl** [comment]
This is better:

    --regex-javascript=/^[ \t]*export[ \t]\{1,\}class[ \t]\{1,\}\([A-Za-z0-9_$]\{1,\}\)/\1/C,Class,Classes/b

This is not a technical problem, really, but a semantic problem: is that a named export or class definition?

**Comment by evanleck** [comment]
Makes sense, thanks! And it’s really both. I have a single file with a single class definition that it exports as a named export (i.e. not the default export).

**Comment by tom3q** [comment]
Hey Romain, thanks a lot for creating this super useful setup! I stumbled across the same problem. Do you think it would be possible to add the regex from your comment above to the repository?

**Comment by romainl** [comment]
@tom3q I'm still not sure how to classify that thing: "class definition" or "named export", as it's both and I have no desire to have dupes in the generated `tags` file. Either way, you could add it to your own fork in the mean time.

**Comment by tom3q** [comment]
Okay, thanks for explanation. I'll keep it in my fork.

**Comment by fabriziobertoglio1987** [comment]
```javascript
import Data from './data';
import Unit from './unit';

export default class Daily extends Data { 
  constructor({ swellHeight, waveHeight, swellPeriod, windSpeed, imperial, i}) {
    super({swellHeight, swellPeriod, windSpeed, imperial})
    this._swellHeight = new Unit({ value: swellHeight[i] })
    this._waveHeight = new Unit({ value: waveHeight[i] })
    this._windSpeed = new Unit({ value: windSpeed[i] })
  }

  get waveHeight() {
    if (this._imperial) {
      return this._waveHeight.ft
    }
    return this._waveHeight.mt
  }
}
```

but if I remove the export it works. 

none of the above solution from @romainl and @evanleck worked.
